src/main/java/com/eerussianguy/firmalife/blocks/BlockFruitDoor.java

```diff
@@ -0,0 +1,50 @@
+package com.eerussianguy.firmalife.blocks;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import com.eerussianguy.firmalife.items.ItemFruitDoor;
+import net.dries007.tfc.api.types.IFruitTree;
+import net.minecraft.block.BlockDoor;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Items;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.World;
+
+public class BlockFruitDoor extends BlockDoor {
+
+    private static final Map<IFruitTree, BlockFruitDoor> MAP = new HashMap<>();
+
+    public static BlockFruitDoor get(IFruitTree tree) { return MAP.get(tree); }
```

Comment:
Is this method used? We have been deliberately avoiding collecting stuff in hashmaps like this. Looks like copypasta from TFC

Reply:
If we don't need the map, we also don't have to construct fruit tree blocks using IFruitTree (and therefore we can construct a cinnamon variant easily)

Reply:
I'm not 100% sure why tfc goes through the motions with the hashmaps, but it seems useful for iterating over groupings of specific items.  You're right that they aren't used right now though